app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperFileManager.kt

```diff
@@ -49,17 +50,126 @@ class WallpaperFileManager(
         }
 
     /**
-     * Remove all wallpapers that are not the [currentWallpaper] or in [availableWallpapers].
+     * Get the list of migrated legacy wallpapers.
+     *
+     * @param remoteAvailableWallpapers List of wallpapers available remote.
+     * @param localAvailableWallpapers List of wallpapers available locally.
+     * @return List of migrated wallpapers.
      */
-    suspend fun clean(currentWallpaper: Wallpaper, availableWallpapers: List<Wallpaper>) = withContext(Dispatchers.IO) {
-        scope.launch {
-            val wallpapersToKeep = (listOf(currentWallpaper) + availableWallpapers).map { it.name }
-            wallpapersDirectory.listFiles()?.forEach { file ->
-                if (file.isDirectory && !wallpapersToKeep.contains(file.name)) {
-                    file.deleteRecursively()
+    suspend fun getMigratedWallpapers(
+        remoteAvailableWallpapers: List<Wallpaper>,
+        localAvailableWallpapers: List<Wallpaper>,
+    ) = withContext(Dispatchers.IO) {
+        File(storageRootDirectory, "wallpapers").listFiles()
+            ?.filter { file ->
```

Comment:
It might make sense to put the migration code into a specialized `WallpaperLegacyMigration` class, so that the whole thing can be easily deleted once we're satisfied that enough time has passed.